frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/IncrementalMojoHelper.java

```diff
@@ -0,0 +1,403 @@
+package com.github.eirslett.maven.plugins.frontend.mojo;
+
+import org.apache.maven.plugin.logging.Log;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IncrementalMojoHelper {
+    private final File workingDirectory;
+    private final Log log;
+    private final boolean isActive;
+
+    public IncrementalMojoHelper(String activationFlag, File workingDirectory, Log log) {
+        this.workingDirectory = workingDirectory;
+        this.log = log;
+
+        this.isActive = activationFlag != null && activationFlag.equals("true");
+    }
+
+    public boolean shouldExecute() {
+        if (!isActive) {
+            return true;
```

Comment:
Why should the incremental helper execute if it is not active?

Reply:
The question is answers is "should the guarded feature execute or be skipped" so yes - if it's not active it responds "yes - please execute this time" :)